From 52c1f61f7858c52c83d78dbfbf92451c8eb911db Mon Sep 17 00:00:00 2001
From: Yongxin Liu <yongxin.liu@windriver.com>
Date: Thu, 13 Jun 2024 10:26:55 +0000
Subject: [PATCH 17/24] rdk/snr: change DEFINE_SEMAPHORE() to take an argument

kernel commit 48380368dec1 ("Change DEFINE_SEMAPHORE() to take a number
argument").

Upstream-Status: Inappropriate [RDK is released in tarball only]

Signed-off-by: Yongxin Liu <yongxin.liu@windriver.com>
---
 klm/qat/qat/drivers/crypto/qat/qat_common/adf_dev_processes.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/klm/qat/qat/drivers/crypto/qat/qat_common/adf_dev_processes.c b/klm/qat/qat/drivers/crypto/qat/qat_common/adf_dev_processes.c
index e356b5e..d8eaf2b 100644
--- a/klm/qat/qat/drivers/crypto/qat/qat_common/adf_dev_processes.c
+++ b/klm/qat/qat/drivers/crypto/qat/qat_common/adf_dev_processes.c
@@ -60,7 +60,11 @@ static struct adf_chr_drv_info adf_processes_drv_info = {
 };
 
 static LIST_HEAD(processes_list);
+#if KERNEL_VERSION(6, 4, 0) > LINUX_VERSION_CODE
 static DEFINE_SEMAPHORE(processes_list_sema);
+#else
+static DEFINE_SEMAPHORE(processes_list_sema, 1);
+#endif
 
 static void adf_chr_drv_destroy(void)
 {
-- 
2.39.3

